app/core/event_handler.py

```diff
@@ -10,6 +12,54 @@
 log = logging.get_logger(__name__, config=config)
 
 
+def determine_original_pid(s3_object_key: str) -> Union[str, None]:
+    """From an `s3_object_key`, determine the original pid.
+
+    For exported items, the "export name", and thus the `s3_object_key` by
+    which the item is uploaded to S3 is `<original_pid>.<extension>`. This
+    "original_pid" can have several forms:
+        - 10 char alphanumeric lowercase string, or,
+        - 10 char alphanumeric lowercase string followed by an underscore and a
+          variant (eg. `mezanine`),
+    As to be not too restrictive, we allow for some variation in the part that
+    follows the underscore. The `pid` itself, however, should not deviate from
+    a 10 char lowercase alphanum string."""
+    pattern = re.compile("^([a-z0-9]{10})_?([a-zA-Z]{0,12})")
```

Comment:
Any reason for using extra capturing groups in the regex? I don't see them used so it could be `^[a-z0-9]{10}_?[a-zA-Z]{0,12}`

Reply:
No reason really. It doesn't change the match but it provides two groups: one for the pid, one for the "variation". Granted, these groups are currently not used. Keep or toss? :-)

Reply:
Resolved. :+1: